lib/rubocop/cop/mixin/uncommunicative_name.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality shared by Uncommunicative cops
+    module UncommunicativeName
+      CASE_MSG = 'Only use lowercase characters for %<name_type>s.'.freeze
+      NUM_MSG = 'Do not end %<name_type>s with a number.'.freeze
+      LENGTH_MSG = '%<name_type>s must be longer than %<min>s ' \
+                   'characters.'.freeze
+
+      def check(node, args)
+        args.each do |arg|
+          source = arg.source
+          case_offense(node, arg) if uppercase?(source)
+          num_offense(node, arg) if ends_with_num?(source)
+          length_offense(node, arg) unless long_enough?(source)
+        end
+      end
+
+      def case_offense(node, arg)
+        add_offense(node, location: arg.source_range,
+                          message: format(CASE_MSG, name_type: name_type(node)))
+      end
+
+      def uppercase?(name)
+        name =~ /[A-Z]/
```

Comment:
Probably it should be `[[:upper:]]`.
 See https://github.com/bbatsov/rubocop/pull/5017

Reply:
Excellent! Thank you!